mentorship ios/Networking/RelationRequestActions.swift

```diff
@@ -40,18 +49,15 @@ class RelationRequestActionAPI: ObservableObject {
         }
         
         //api call
-        cancellable = NetworkManager.callAPI(urlString: urlString, httpMethod: httpMethod, token: token)
+        cancellable = NetworkManager.callAPI(urlString: urlString, httpMethod: httpMethod, token: token, session: urlSession)
             .receive(on: RunLoop.main)
             .catch { _ in Just(self.response) }
-            .sink { [weak self] in
-                self?.response = $0
-                if NetworkManager.responseCode == 200 {
-                    self?.success = true
-                }
+            .sink {
+                completion($0)
         }
     }
 }
 
-struct ResponseMessage: Decodable {
+struct ResponseMessage: Codable {
```

Comment:
This should be your `RequestActionViewModel`

Reply:
I don't feel confident about this. This is the data model that is used to decode and map the server response data. While it is a single property here, this is the pattern that has been followed throughout the app.

Reply:
Yeah you can keep middleware data class, but then you need a mapper to map from DataModel to ViewModel. And please rename it respectively so it contain Model in name.